android/src/com/mapswithme/maps/settings/StoragePathManager.java

```diff
@@ -517,18 +374,10 @@ private int changeStorage(StorageItem newStorage, StorageItem oldStorage)
     {
       for (int i = 0; i < oldFiles.length; ++i)
       {
-        if (!MapStorage.nativeMoveFile(oldFiles[i].getAbsolutePath(), newFiles[i].getAbsolutePath()))
-        {
-          File parent = newFiles[i].getParentFile();
-          if (parent != null)
-            parent.mkdirs();
-          StorageUtils.copyFile(oldFiles[i], newFiles[i]);
-        }
-        else
-        {
-          // No need to delete oldFiles[i] because it was moved to newFiles[i].
-          oldFiles[i] = null;
-        }
+        File parent = newFiles[i].getParentFile();
+        if (parent != null)
+          parent.mkdirs();
+        StorageUtils.copyFile(oldFiles[i], newFiles[i]);
```

Comment:
MapStorage.nativeMoveFile() will never succeed for storages on different mount points.